golang/cosmos/x/swingset/genesis.go

```diff
@@ -169,32 +190,72 @@ type swingStoreGenesisEventHandler struct {
 	snapshotHeight uint64
 	swingStore     sdk.KVStore
 	hasher         hash.Hash
+	exportMode     string
 }
 
 func (eventHandler swingStoreGenesisEventHandler) OnExportStarted(height uint64, retrieveSwingStoreExport func() error) error {
 	return retrieveSwingStoreExport()
 }
 
 func (eventHandler swingStoreGenesisEventHandler) OnExportRetrieved(provider keeper.SwingStoreExportProvider) error {
-	if eventHandler.snapshotHeight != provider.BlockHeight {
+	if !((eventHandler.exportMode == keeper.SwingStoreArtifactModeDebug && eventHandler.snapshotHeight == 0) ||
+		eventHandler.snapshotHeight == provider.BlockHeight) {
```

Comment:
Part of this check is technically redundant 

```suggestion
	if eventHandler.exportMode != keeper.SwingStoreArtifactModeDebug &&
		eventHandler.snapshotHeight != provider.BlockHeight {
```